### ISSUE TITLE
SIMD test failed in cctest on hifive unleashed.

### ISSUE THREAD

**Issue by luyahan** [v8-riscv/v8#113]
**Describe the bug**
some simd test error, but we don't implement SIMD.

**To Reproduce**
1. Build with options 'riscv64.native.release'
2. Run test/program 'python2 ./tools/run-tests.py --outdir=./riscv64.native.release/   cctest'
3. See error

**Expected behavior**

**Errors/Logs**
```
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4AddWithI32x4_simd_lowered ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3064
# Check failed: 1 == r.Call() (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3ffff23e78
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ae6b9761e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ae6b0a21e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ae6905512]
    /home/yahan/riscv64.native.release/cctest(+0xedf792) [0x2ae48c8792]
    /home/yahan/riscv64.native.release/cctest(+0xedf7c0) [0x2ae48c87c0]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2ae4279890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2ae4205450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fe0ac1204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2ae4276834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4AddWithI32x4_simd_lowered --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner

                                                                              
[39:10|%  79|+ 5224|-   4]: cctest/test-run-wasm-simd/RunWasm_SimdF32x4AddW...
                                                                              
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4AddWithI32x4_turbofan ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3064
# Check failed: 1 == r.Call() (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3fff976e88
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ae20b161e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ae202421e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ae1e1f512]
    /home/yahan/riscv64.native.release/cctest(+0xedf792) [0x2adfde2792]
    /home/yahan/riscv64.native.release/cctest(+0xedf82c) [0x2adfde282c]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2adf793890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2adf71f450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fe096c204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2adf790834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4AddWithI32x4_turbofan --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner

                                                                                                                                                           
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4ExtractWithI32x4_simd_lowered ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3033
# Check failed: 1 == r.Call() (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3fffc83e88
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ab064961e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ab05bc21e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ab03b7512]
    /home/yahan/riscv64.native.release/cctest(+0xedf446) [0x2aae37a446]
    /home/yahan/riscv64.native.release/cctest(+0xedf474) [0x2aae37a474]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2aadd2b890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2aadcb7450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fd0ecb204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2aadd28834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4ExtractWithI32x4_simd_lowered --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner

                                                                              
[39:16|%  79|+ 5230|-   6]: cctest/test-run-wasm-simd/RunWasm_SimdF32x4Extr...
                                                                              
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4ExtractWithI32x4_turbofan ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3033
# Check failed: 1 == r.Call() (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3fffb6be88
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ad8d0161e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ad8c7421e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ad8a6f512]
    /home/yahan/riscv64.native.release/cctest(+0xedf446) [0x2ad6a32446]
    /home/yahan/riscv64.native.release/cctest(+0xedf4e0) [0x2ad6a324e0]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2ad63e3890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2ad636f450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3ff1bcd204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2ad63e0834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4ExtractWithI32x4_turbofan --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner

                                                                              

                                                                              
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4GetGlobal_simd_lowered ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3247
# Check failed: 1 == r.Call(0) (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3fffae2e78
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ad48c361e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ad483621e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ad4631512]
    /home/yahan/riscv64.native.release/cctest(+0xec7972) [0x2ad25dc972]
    /home/yahan/riscv64.native.release/cctest(+0xec79a0) [0x2ad25dc9a0]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2ad1fa5890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2ad1f31450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fc606c204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2ad1fa2834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4GetGlobal_simd_lowered --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner

                                                                              
=== cctest/test-run-wasm-simd/RunWasm_SimdF32x4GetGlobal_turbofan ===
#
# Fatal error in ../../test/cctest/wasm/test-run-wasm-simd.cc, line 3247
# Check failed: 1 == r.Call(0) (1 vs. 0).
#
#
#
#FailureMessage Object: 0x3fff831e78
==== C stack trace ===============================

    /home/yahan/riscv64.native.release/cctest(+0x31ae61e) [0x2ad338c61e]
    /home/yahan/riscv64.native.release/cctest(+0x312121e) [0x2ad32ff21e]
    /home/yahan/riscv64.native.release/cctest(+0x2f1c512) [0x2ad30fa512]
    /home/yahan/riscv64.native.release/cctest(+0xec7972) [0x2ad10a5972]
    /home/yahan/riscv64.native.release/cctest(+0xec7a0c) [0x2ad10a5a0c]
    /home/yahan/riscv64.native.release/cctest(+0x890890) [0x2ad0a6e890]
    /home/yahan/riscv64.native.release/cctest(+0x81c450) [0x2ad09fa450]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fd289d204]
    /home/yahan/riscv64.native.release/cctest(_start+0x2c) [0x2ad0a6b834]
Command: riscv64.native.release/cctest test-run-wasm-simd/RunWasm_SimdF32x4GetGlobal_turbofan --random-seed=-1744354316 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner                                                                       
```

**Additional context**

**Comment by penguinwu** [comment]
This problem also manifests after the fix in #120 in simulated build after nan-boxing support for float.

Since before the fix #120 the test passes, it may be worth-while to compare the trace-simd before and after the fix in #120 to figure out the difference. Luckily this test consists of only 129 dynamic instructions.

**Comment by penguinwu** [comment]
I will investigate the failure on the simulated build.

**Comment by penguinwu** [comment]
Fix is ready but waiting for #153 to merge in so that the skipped tests can be enabled again.

**Comment by fw-brice** [comment]
I added the additional tests:
```
  'test-run-wasm-simd-scalar-lowering/RunWasm_I8x16ToF32x4_simd_lowered': [SKIP],
  'test-run-wasm-simd/RunWasm_SimdF32x4AddWithI32x4_liftoff: [SKIP],
  'test-run-wasm-simd/RunWasm_SimdF32x4GetGlobal_liftoff: [SKIP],
  'test-run-wasm-simd/RunWasm_SimdF32x4ExtractWithI32x4_liftoff: [SKIP],
```

**Comment by penguinwu** [comment]
Fixed by #156

**Comment by penguinwu** [comment]
@luyahan Please verify that this is fixed by #156, then close the issue.
